### DISCUSSION TITLE
posts/matrix-calculus

### DISCUSSION THREAD

**Discussion by giscus** [acciochris/acciochris.github.io#13] [Announcements]
# posts/matrix-calculus

https://acciochris.github.io/posts/matrix-calculus.html

**Comment by acciochris** [comment]
I apparently forgot about gradient descent 😥.

After figuring out the derivative of the loss function w.r.t. the weights and biases, we will need to update them so that the loss can decrease. In the example above, the updated bias $\mathbf{b}'_h$ should be

$$
\mathbf{b}'_h = \mathbf{b}_h - \eta\dfrac{\partial L}{\partial \mathbf{b}_h}
$$

where $\eta$ is the learning rate.

**Comment by acciochris** [comment]
Another thing to mention.

IIRC, I have followed PyTorch's convention of putting the `n_samples` dimension in the front, which makes the bias vectors row vectors, which in turn makes the dimension of Jacobians a bit weird. If anyone knows what's going on here with the dimensions, please reply directly to this thread. Thanks!
